common/src/sql/dbinit.sql

```diff
@@ -2160,6 +2160,258 @@ CREATE TABLE omicron.public.role_assignment (
      )
 );
 
+/*******************************************************************/
+
+/*
+ * External Networking
+ *
+ * **For more details on external networking see RFD 267**
+ */
+
+CREATE TYPE omicron.public.address_lot_kind AS ENUM (
+    'infra',
+    'pool'
+);
+
+CREATE TABLE omicron.public.address_lot (
+    id UUID PRIMARY KEY,
+    name STRING(63) NOT NULL,
+    description STRING(512) NOT NULL,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    time_deleted TIMESTAMPTZ,
+    kind omicron.public.address_lot_kind NOT NULL
+);
+
+CREATE UNIQUE INDEX ON omicron.public.address_lot (
+    name
+) WHERE
+    time_deleted IS NULL;
+
+CREATE TABLE omicron.public.address_lot_block (
+    id UUID PRIMARY KEY,
+    address_lot_id UUID NOT NULL,
+    first_address INET NOT NULL,
+    last_address INET NOT NULL
+);
+
+CREATE INDEX ON omicron.public.address_lot_block (
+    address_lot_id
+);
+
+CREATE TABLE omicron.public.address_lot_rsvd_block (
+    id UUID PRIMARY KEY,
+    address_lot_id UUID NOT NULL,
+    first_address INET NOT NULL,
+    last_address INET NOT NULL
+);
+
+CREATE INDEX ON omicron.public.address_lot_rsvd_block (
+    address_lot_id
+);
+
+CREATE TABLE omicron.public.loopback_address (
+    id UUID PRIMARY KEY,
+    time_created TIMESTAMPTZ NOT NULL,
+    time_modified TIMESTAMPTZ NOT NULL,
+    address_lot_block_id UUID NOT NULL,
+    rsvd_address_lot_block_id UUID NOT NULL,
+    rack_id UUID NOT NULL,
+    switch_location TEXT NOT NULL,
+    address INET NOT NULL
+);
+
+CREATE UNIQUE INDEX ON omicron.public.loopback_address (
+    address, rack_id, switch_location
+);
```

Comment:
I've mentioned this in other places in the PR, but we're not going to want to associate switch resources with `rack_id` and `switch_location`. For the sake of reducing churn for this PR we can merge it in this state so long as we follow up on creating a top level switch resource and changing all the `rack_id` + `switch_location` pairs to map to `switch_id`.